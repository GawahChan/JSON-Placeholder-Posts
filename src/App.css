.app {
    width: 100%;
    height: 100%;
}

.app-body {
    display: flex;
    flex-direction: column-reverse;
    padding: 2rem 1rem;
}

.all-posts {
    flex: 1;
    padding: 10px;
}

.my-posts {
    flex: 1;
    padding: 10px;
} 

@media only screen and (min-width: 768px) {
    .app-body {
        flex-direction: row;
    }
    .header {
        padding: 3rem 1rem;
    }
}

@media only screen and (min-width: 2560px) {
    .header {
        padding: 6rem 2rem;
    }
}