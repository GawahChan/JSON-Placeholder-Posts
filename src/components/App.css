.app-body {
    display: flex
}

.all-posts {
    flex: 1
}

.my-posts {
    flex: 1
}