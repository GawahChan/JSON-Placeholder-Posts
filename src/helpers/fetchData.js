function fetchData() {
    return (
        fetch('https://jsonplaceholder.typicode.com/posts')
    );
}

export default fetchData;