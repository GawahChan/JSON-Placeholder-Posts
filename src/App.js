import React, { Component } from 'react';
import Header from './components/Header';
import Post from './components/Post';
import SubTitle from './components/SubTitle';
import fetchData from './helpers/fetchData';
import inputs from './helpers/inputs';


import './App.css';

class App extends Component {
    constructor() {
        super();
        this.state = {
            posts: [],
            myPost: [],
            inputValue: '',
            displayAllPosts: false,
            displayMyPosts: true
        };
    }

    componentDidMount() {
        fetchData()
            .then(res => res.json())
            .then(posts => this.setState({ posts }))
    }

    filterPosts = (inputValue) => {
        const myPost = [];
        const input = inputs(inputValue);
        this.state.posts.filter(post => {
            return input.forEach(number => {
                return post.userId === number ? myPost.push(post) : null
            });
        });
        this.setState({ inputValue, myPost })
    };

    toggleDisplayAllPosts = () => {
        this.setState({ displayAllPosts: !this.state.displayAllPosts })
    }

    toggleDisplayMyPosts = () => {
        this.setState({ displayMyPosts: !this.state.displayMyPosts })
    }

    render() {
        return (
            <div className='app'>
                <Header title='JSON-Placeholder Posts' inputValue={this.state.inputValue} filterPosts={this.filterPosts} />
                <div className='app-body'>
                    <div className='all-posts'>
                        <SubTitle title='All Posts' display={this.state.displayAllPosts} toggleDisplay={this.toggleDisplayAllPosts} />
                        { 
                            this.state.displayAllPosts && this.state.posts.map(post => {
                                return (
                                    <Post
                                        key={post.id}
                                        userId={post.userId}
                                        postId={post.id}
                                        postTitle={post.title}
                                        postBody={post.body}
                                    />
                                );
                            })
                        }
                    </div>
                    <div className='my-posts'>
                        <SubTitle title='My Posts' display={this.state.displayMyPosts} toggleDisplay={this.toggleDisplayMyPosts} />
                        {
                            this.state.displayMyPosts && this.state.myPost.map(post => {
                                return (
                                    <Post
                                        key={post.id}
                                        userId={post.userId}
                                        postId={post.id}
                                        postTitle={post.title}
                                        postBody={post.body}
                                    />
                                );
                            })
                        }
                    </div>
                </div>
            </div>
        );
    }
}

export default App;
