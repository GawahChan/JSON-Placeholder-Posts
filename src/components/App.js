import React, { Component } from 'react';
import Header from './Header';
import Post from './Post';
import fetchData from '../helpers/fetchData';

import './App.css';

class App extends Component {
    constructor() {
        super();
        this.state = {
            posts: [],
            myPost: [],
            inputValue: '',
        };
    }

    componentDidMount() {
        fetchData()
            .then(res => res.json())
            .then(posts => this.setState({ posts }))
    }

    filterPosts = (inputValue) => {
        const myPost = this.state.posts.filter(post => post.userId === Number(inputValue));
        this.setState({ inputValue, myPost })
    };

    render() {
        return (
            <div className='app'>
                <Header title='Posts Page' inputValue={this.state.inputValue} filterPosts={this.filterPosts} />
                <div className='app-body'>
                    <div className='all-posts'>
                        <h2>All posts</h2>
                        {
                            this.state.posts.map(post => {
                                return (
                                    <Post
                                        key={post.id}
                                        userId={post.userId}
                                        postId={post.id}
                                        postTitle={post.title}
                                        postBody={post.body}
                                    />
                                );
                            })
                        }
                    </div>
                    <div className='my-posts'>
                        <h2>My posts</h2>
                        {
                            this.state.myPost.map(post => {
                                return (
                                    <Post
                                        key={post.id}
                                        userId={post.userId}
                                        postId={post.id}
                                        postTitle={post.title}
                                        postBody={post.body}
                                    />
                                );
                            })
                        }
                    </div>
                </div>
            </div>
        );
    }
}

export default App;
