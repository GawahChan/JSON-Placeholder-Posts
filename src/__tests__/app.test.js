import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../components/App';
import fetchData from '../helpers/fetchData';


describe('App', () => {
    const appTree = renderer.create(<App />).toJSON();
    const app = shallow(<App />);

    it('renders correctly', () => {
        expect(appTree).toMatchSnapshot();
    });

    it('initialises state `posts` with an empty array', () => {
        expect(app.state().posts).toEqual([]);
    });

    it('initialises state `myPost` with an empty array', () => {
        expect(app.state().myPost).toEqual([]);
    });

    it('initialises state `inputValue` with an empty string', () => {
        expect(app.state().inputValue).toEqual('');
    });

    describe('componentDidMount', () => {
        const spy = jest.spyOn(App.prototype, 'componentDidMount');

        beforeEach(() => {
            shallow(<App />)
        })
        afterEach(() => {
            app.setState({ posts: [] });
        })

        it('calls componentDidMount', () => {
            expect(spy).toHaveBeenCalledTimes(1);
        });

        it('updates state `posts` with data and checks if it re-renders correctly', () => {
            return fetchData()
                .then(res => {
                    expect(res).toBeDefined();
                    expect(res.status).toEqual(200);
                    return res.json()
                })
                .then(posts => {
                    app.setState({ posts })
                    expect(app.state().posts).toEqual(posts);
                    expect(app.find('Post').exists()).toBe(true);
                });
        });
    });

    // describe('filterPosts()', () => {

    //     it('checks if function is defined', () => {
    //         expect(app.instance().filterPosts).toBeDefined();
    //     });

    //     it('test,', () => {
    //         app.instance().filterPosts
    //     })
    // });

    describe('render', () => {
        it('render `Header` component', () => {
            expect(app.find('Header').exists()).toBe(true);
        });

        it('renders `.all-posts` section', () => {
            expect(app.find('.all-posts').children().length).toBeGreaterThanOrEqual(1);
        });

        it('renders `.my-posts` section', () => {
            expect(app.find('.my-posts').children().length).toBeGreaterThanOrEqual(1);
        });
    });
});