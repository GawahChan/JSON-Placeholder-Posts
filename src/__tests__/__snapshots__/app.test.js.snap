// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="app"
>
  <div>
    <h1>
      Posts Page
    </h1>
    <input
      className="input-filterPosts"
      onChange={[Function]}
      placeholder="Please select between 1 and 10"
      value=""
    />
  </div>
  <div
    className="app-body"
  >
    <div
      className="all-posts"
    >
      <h2>
        All posts
      </h2>
    </div>
    <div
      className="my-posts"
    >
      <h2>
        My posts
      </h2>
    </div>
  </div>
</div>
`;
